python/semantic_kernel/connectors/ai/function_choice_behavior.py

```diff
@@ -0,0 +1,281 @@
+            based on either the specified filters or the full qualified names.
+
+    """
+
+    enable_kernel_functions: bool = True
+    maximum_auto_invoke_attempts: int = DEFAULT_MAX_AUTO_INVOKE_ATTEMPTS
+    filters: (
+        dict[Literal["excluded_plugins", "included_plugins", "excluded_functions", "included_functions"], list[str]]
+        | None
+    ) = None
+    function_fully_qualified_names: list[str] | None = None
+    type: FunctionChoiceType | None = None
+
+    @classmethod
+    def from_function_call_behavior(cls, behavior: "FunctionCallBehavior") -> "FunctionChoiceBehavior":
+        """Create a FunctionChoiceBehavior from a FunctionCallBehavior."""
+        from semantic_kernel.connectors.ai.function_call_behavior import (
+            EnabledFunctions,
+            KernelFunctions,
+            RequiredFunction,
+        )
+
+        if isinstance(behavior, (EnabledFunctions, KernelFunctions)):
+            return cls.Auto(
+                auto_invoke=behavior.auto_invoke_kernel_functions,
+                filters=behavior.filters if hasattr(behavior, "filters") else None,
+            )
+        if isinstance(behavior, (RequiredFunction)):
+            return cls.Required(
+                auto_invoke=behavior.auto_invoke_kernel_functions,
+                function_fully_qualified_names=[behavior.function_fully_qualified_name]
+                if hasattr(behavior, "function_fully_qualified_name")
+                else None,
+            )
+        return cls(
+            enable_kernel_functions=behavior.enable_kernel_functions,
+            maximum_auto_invoke_attempts=behavior.max_auto_invoke_attempts,
+        )
+
+    @property
+    def auto_invoke_kernel_functions(self):
+        """Return True if auto_invoke_kernel_functions is enabled."""
+        return self.maximum_auto_invoke_attempts > 0
+
+    @auto_invoke_kernel_functions.setter
+    def auto_invoke_kernel_functions(self, value: bool):
+        """Set the auto_invoke_kernel_functions property."""
+        self.maximum_auto_invoke_attempts = DEFAULT_MAX_AUTO_INVOKE_ATTEMPTS if value else 0
+
+    def _check_and_get_config(
+        self, kernel: "Kernel", fully_qualified_names: list[str] | None = None, filters: dict[str, Any] | None = {}
+    ) -> FunctionCallChoiceConfiguration:
+        """Check for missing functions and get the function call choice configuration."""
+        full_kernel_function_metadata = set(kernel.get_full_list_of_function_metadata())
+
+        if fully_qualified_names:
+            _check_for_missing_functions(fully_qualified_names, full_kernel_function_metadata)
+            return FunctionCallChoiceConfiguration(
+                available_functions=kernel.get_list_of_function_metadata({"included_functions": fully_qualified_names})
+            )
+        if filters:
+            return FunctionCallChoiceConfiguration(available_functions=kernel.get_list_of_function_metadata(filters))
```

Comment:
how does it handle the list of strings?

Reply:
When you say a list of strings, you're referring to someone configuring the FunctionChoiceBehavior to be used with `time-time` or `math-Add`? That can be done either with the `filters` or the `function_fully_qualified_names` param. The `function_fully_qualified_names` are used to map the specified functions in the yaml/JSON config to the FunctionChoiceBehavior.

Now that I look at it more, it seems like we could get rid of `function_fully_qualified_names` and solely use `filters` -- where if `time-time` or `math-Add` is specified in the config, we create the proper filter from it. It's then one less thing in code to have to do a check on, or worry about.